rs-matter/src/data_model/subscriptions.rs

```diff
@@ -0,0 +1,142 @@
+/*
+ *
+ *    Copyright (c) 2020-2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+use core::cell::RefCell;
+
+use embassy_sync::blocking_mutex::raw::NoopRawMutex;
+use embassy_time::Instant;
+
+use portable_atomic::{AtomicU32, Ordering};
+
+use crate::utils::notification::Notification;
+
+struct Subscription {
+    node_id: u64,
+    id: u32,
+    min_int_secs: u16,
```

Comment:
I wonder if a comment of `we use u16 instead of embassy::Duration to save some storage` would be worth it (to show this was intentional).

we could even have a `pub fn min_interfval(&self) -> embassy_time::Duration` to use to make our lines shorter.

Reply:
Comment is in place. `min_interval` I would introduce only when these `u16`s become used and converted to `embassy_time::Duration` in more than one place, which is currently not the case. I mean, we already have `report_due` and `is_expired`